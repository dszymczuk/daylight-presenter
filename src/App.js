import React, { useContext } from 'react';
import { Col, Container, Row } from "react-bootstrap";
import styles from './App.module.scss';
import Location from "./components/location";
import LocationForm from "./form/location";
import LocationContextProvider, { Location as LocationContext } from "./contexts/locations";
import ErrorBoundary from "./components/errorBoundary";


const App = () => {
  const locationContext = useContext(LocationContext);
  const { locations } = locationContext;

  return (
    <div className={styles.appWrapper}>
      <Container>
        {locations.map((location, index) => {
          return (
            <Row key={index}>
              <Col>
                <ErrorBoundary>
                  <Location {...location} index={index}/>
                </ErrorBoundary>
              </Col>
            </Row>
          )
        })
        }
        <Row>
          <Col>
            <LocationForm/>
          </Col>
        </Row>
      </Container>
    </div>
  );
};

const AppWrapper = () => (
  <LocationContextProvider>
    <App/>
  </LocationContextProvider>
);

export default AppWrapper;
