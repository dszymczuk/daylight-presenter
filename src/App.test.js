import React from "react";
import { render, cleanup } from '@testing-library/react';
import App from "./App";

afterEach(cleanup);

it("renders App", () => {
  const { asFragment } = render(<App/>);
  expect(asFragment()).toMatchSnapshot();
});