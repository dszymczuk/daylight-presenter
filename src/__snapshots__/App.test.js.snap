// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App 1`] = `
<DocumentFragment>
  <div
    class="appWrapper"
  >
    <div
      class="container"
    >
      <div
        class="row"
      >
        <div
          class="col"
        >
          <div
            class="card"
          >
            <h5
              class="card-header"
            >
              <div
                class="header"
              >
                <div>
                  Krak√≥w 
                  <span>
                    (50.064651, 19.944981)
                  </span>
                </div>
                <div>
                  24.01.2020
                  <button
                    class="btn btn-warning btn-sm"
                    type="button"
                  >
                    Edit
                  </button>
                  <button
                    class="btn btn-danger btn-sm"
                    type="button"
                  >
                    Remove
                  </button>
                </div>
              </div>
            </h5>
            <div
              class="card-body"
            >
              <div
                class="progressBarWrapper"
              >
                <div
                  class="progress"
                >
                  <div
                    aria-valuemax="1440"
                    aria-valuemin="0"
                    aria-valuenow="NaN"
                    class="progress-bar bg-warning"
                    role="progressbar"
                  />
                  <div
                    aria-valuemax="1440"
                    aria-valuemin="0"
                    aria-valuenow="NaN"
                    class="progress-bar bg-success"
                    role="progressbar"
                  />
                  <div
                    aria-valuemax="1440"
                    aria-valuemin="0"
                    aria-valuenow="NaN"
                    class="progress-bar bg-warning"
                    role="progressbar"
                  />
                </div>
                <div
                  class="labels"
                >
                  <div
                    class="startHour"
                  >
                    00:00
                  </div>
                  <div
                    class="endHour"
                  >
                    23:59
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col"
        >
          <div
            class="card"
          >
            <h5
              class="card-header"
            >
              <div
                class="header"
              >
                <div>
                  Tokio 
                  <span>
                    (39.758602, -104.997437)
                  </span>
                </div>
                <div>
                  25.01.2020
                  <button
                    class="btn btn-warning btn-sm"
                    type="button"
                  >
                    Edit
                  </button>
                  <button
                    class="btn btn-danger btn-sm"
                    type="button"
                  >
                    Remove
                  </button>
                </div>
              </div>
            </h5>
            <div
              class="card-body"
            >
              <div
                class="progressBarWrapper"
              >
                <div
                  class="progress"
                >
                  <div
                    aria-valuemax="1440"
                    aria-valuemin="0"
                    aria-valuenow="NaN"
                    class="progress-bar bg-warning"
                    role="progressbar"
                  />
                  <div
                    aria-valuemax="1440"
                    aria-valuemin="0"
                    aria-valuenow="NaN"
                    class="progress-bar bg-success"
                    role="progressbar"
                  />
                  <div
                    aria-valuemax="1440"
                    aria-valuemin="0"
                    aria-valuenow="NaN"
                    class="progress-bar bg-warning"
                    role="progressbar"
                  />
                </div>
                <div
                  class="labels"
                >
                  <div
                    class="startHour"
                  >
                    00:00
                  </div>
                  <div
                    class="endHour"
                  >
                    23:59
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col"
        >
          <div
            class="card"
          >
            <h5
              class="card-header"
            >
              <div
                class="header"
              >
                <div>
                  New York 
                  <span>
                    (55.755825, 37.617298)
                  </span>
                </div>
                <div>
                  25.01.2020
                  <button
                    class="btn btn-warning btn-sm"
                    type="button"
                  >
                    Edit
                  </button>
                  <button
                    class="btn btn-danger btn-sm"
                    type="button"
                  >
                    Remove
                  </button>
                </div>
              </div>
            </h5>
            <div
              class="card-body"
            >
              <div
                class="progressBarWrapper"
              >
                <div
                  class="progress"
                >
                  <div
                    aria-valuemax="1440"
                    aria-valuemin="0"
                    aria-valuenow="NaN"
                    class="progress-bar bg-warning"
                    role="progressbar"
                  />
                  <div
                    aria-valuemax="1440"
                    aria-valuemin="0"
                    aria-valuenow="NaN"
                    class="progress-bar bg-success"
                    role="progressbar"
                  />
                  <div
                    aria-valuemax="1440"
                    aria-valuemin="0"
                    aria-valuenow="NaN"
                    class="progress-bar bg-warning"
                    role="progressbar"
                  />
                </div>
                <div
                  class="labels"
                >
                  <div
                    class="startHour"
                  >
                    00:00
                  </div>
                  <div
                    class="endHour"
                  >
                    23:59
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col"
        >
          <div>
            <form
              class=""
            >
              <div
                class="card"
              >
                <h5
                  class="card-header"
                >
                  <span>
                    Add new location
                  </span>
                </h5>
                <div
                  class="card-body"
                >
                  <div
                    class="row"
                  >
                    <div
                      class="col"
                    >
                      <div
                        class="form-group"
                      >
                        <label
                          class="form-label"
                          for="name"
                        >
                          Name
                        </label>
                        <input
                          class="form-control"
                          id="name"
                          type="text"
                          value=""
                        />
                      </div>
                    </div>
                    <div
                      class="col"
                    >
                      <div
                        class="form-group"
                      >
                        <label
                          class="form-label"
                          for="lat"
                        >
                          Lat
                        </label>
                        <input
                          class="form-control"
                          id="lat"
                          type="text"
                          value=""
                        />
                      </div>
                    </div>
                    <div
                      class="col"
                    >
                      <div
                        class="form-group"
                      >
                        <label
                          class="form-label"
                          for="lng"
                        >
                          Lng
                        </label>
                        <input
                          class="form-control"
                          id="lng"
                          type="text"
                          value=""
                        />
                      </div>
                    </div>
                    <div
                      class="col"
                    >
                      <div
                        class="datepickerWrapper"
                      >
                        <div
                          class="form-group"
                        >
                          <label
                            class="form-label"
                            for="date"
                          >
                            Date
                          </label>
                          <div
                            class="react-datepicker-wrapper"
                          >
                            <div
                              class="react-datepicker__input-container"
                            >
                              <input
                                class="form-control"
                                id="date"
                                value="26.01.2020"
                              />
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="row"
                  >
                    <div
                      class="col"
                    >
                      <button
                        class="btn btn-success btn-block"
                        type="submit"
                      >
                        Add location
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
